src/libraries/System.Private.CoreLib/src/System/Numerics/Matrix4x4.cs

```diff
@@ -2010,7 +2071,53 @@ public static unsafe Matrix4x4 Lerp(Matrix4x4 matrix1, Matrix4x4 matrix2, float
         /// <returns>The result of the multiplication.</returns>
         public static unsafe Matrix4x4 operator *(Matrix4x4 value1, Matrix4x4 value2)
         {
-            if (Sse.IsSupported)
+            if (AdvSimd.Arm64.IsSupported)
+            {
+                Unsafe.SkipInit(out Matrix4x4 result);
+
+                // Perform the operation on the first row
+
+                Vector128<float> M11 = AdvSimd.LoadVector128(&value1.M11);
+
+                Vector128<float> vX = AdvSimd.MultiplyBySelectedScalar(AdvSimd.LoadVector128(&value2.M11), M11, 0);
+                Vector128<float> vY = AdvSimd.MultiplyBySelectedScalar(AdvSimd.LoadVector128(&value2.M21), M11, 1);
+                Vector128<float> vZ = AdvSimd.Arm64.FusedMultiplyAddBySelectedScalar(vX, AdvSimd.LoadVector128(&value2.M31), M11, 2);
+                Vector128<float> vW = AdvSimd.Arm64.FusedMultiplyAddBySelectedScalar(vY, AdvSimd.LoadVector128(&value2.M41), M11, 3);
+
+                AdvSimd.Store(&result.M11, AdvSimd.Add(vZ, vW));
+
+                // Repeat for the other 3 rows
+
+                Vector128<float> M21 = AdvSimd.LoadVector128(&value1.M21);
```

Comment:
Lot of repeated code. consider extracting the code in a function and passing 4 rows data to it.

Reply:
The matrix code is small and is more susceptible to inlining differences than some. I can investigate extracting it for .NET 6